### PR TITLE
[REVIEW] Add dictionary support to cudf::minmax

### PR DESCRIPTION
**Pull request by davidwendt**
Reference #5963

Add support for dictionary column types in `cudf::minmax`. This PR also adds support for strings column types as well. The code was refactored to re-use the internal `reduce_device` function for all supported types. The strings specialization logic just creates a `string_scalar` result which requires a copy step to build a new `string_scalar` object from a `string_view` (output from the `reduce_device` function). The dictionary support was to just to substitute the `keys()` column in the detail function right before the type-dispatcher call.

Added a gtests for strings as well as dictionary with strings keys.

### CODE REVIEW

**Review by devavret — APPROVED** [review]

File: cpp/src/reductions/minmax.cu
Comment (devavret) [inline cpp/src/reductions/minmax.cu:239]: This won't work for sliced dictionary columns right? The keys would be for the entire base column but the actual view of the column for which to get the minmax would be described by the indices and the offset applied, if any.

**Review by devavret — COMMENTED** [review]
Just realized something. Not requesting changes because need confirmation if what I'm thinking is correct.

File: cpp/src/reductions/minmax.cu
Comment (davidwendt) [reply cpp/src/reductions/minmax.cu:239]: Yes, you are right. Good catch.

File: cpp/src/reductions/minmax.cu
Comment (devavret) [inline cpp/src/reductions/minmax.cu:250]: Might want to update this to use stream view as per Mark's PR

**Review by devavret — APPROVED** [review]
One non-dealbreaker comment but approve otherwise.

**Review by trxcllnt — APPROVED** [review]

**Comment by harrism** [comment]
Please hold off  on merging until #6648 is merged (just waiting on final CI on that one, should be soon). Each PR that is merged is delaying that one due to long CI times.

**Pull request merged**
